src/api/routers/vertex.py

```diff
@@ -0,0 +1,195 @@
+
+def get_header(model, request, path):
+    path_no_prefix = f"/{path.lstrip('/')}".removeprefix(API_ROUTE_PREFIX)
+    target_url = get_gcp_target(model, path_no_prefix)
+
+    # remove hop-by-hop headers
+    headers = {
+        k: v for k, v in request.headers.items()
+        if k.lower() not in {"host", "content-length", "accept-encoding", "connection", "authorization"}
+    }
+
+    # Fetch service account token
+    access_token = get_access_token()
+    headers["Authorization"] = f"Bearer {access_token}"
+    return target_url, headers
+
+def to_vertex_anthropic(openai_messages):
+    message = [
+        {
+            "role": m["role"],
+            "content": [{"type": "text", "text": m["content"]}]
+        }
+        for m in openai_messages["messages"]
+    ]
+
+    return {
+        "anthropic_version": "vertex-2023-10-16",
+        "max_tokens": 256,
+        "messages": message
+    }
+
+def from_anthropic_to_openai_response(msg):
+    msg_json = json.loads(msg)
+    return json.dumps({
+        "id": msg_json["id"],
+        "object": "chat.completion",
+        "model": msg_json.get("model", "claude"),
+        "choices": [
+            {
+                "index": 0,
+                "message": {
+                    "role": msg_json["role"],
+                    "content": "".join(
+                        part["text"] for part in msg_json["content"]
+                        if part["type"] == "text"
+                    )
+                },
+                "finish_reason": msg_json.get("stop_reason", "stop")
+            }
+        ],
+        "usage": msg_json.get("usage", {})
+    })
+
+async def handle_proxy(request: Request, path: str):
+    try:
+        content = await request.body()
+        content_json = json.loads(content)
+
+        if USE_MODEL_MAPPING:
+            if "model" in content_json:
+                request_model = content_json.get("model", None)
+                model = get_model("gcp", request_model)
+                model_name = model
+
+                if model != None and model != request_model and "publishers/google/" in model:
+                    model_name = f"google/{model.split('/')[-1]}"
+
+                content_json["model"]= model_name
+
+        needs_conversion = False
+        if not model in known_chat_models:
+            needs_conversion = True
+            # openai messages to vertex contents 
+            if "anthropic" in model:
+                content_json = to_vertex_anthropic(content_json)
+
+        # Build safe target URL
+        target_url, headers = get_header(model, request, path)
+        async with httpx.AsyncClient() as client:
+            response = await client.request(
+                method=request.method,
+                url=target_url,
+                headers=headers,
+                content=json.dumps(content_json),
+                params=request.query_params,
+                timeout=30.0,
```

Comment:
This feels generous. Could we reduce this timeout to 5s or less?
```suggestion
                timeout=5.0,
```